controllers/imagerepository_controller.go

```diff
@@ -918,3 +928,26 @@ func generateDockerconfigSecretData(quayImageURL string, robotAccount *quay.Robo
 		quayImageURL, base64.StdEncoding.EncodeToString([]byte(authString)))
 	return secretData
 }
+
+func (r *ImageRepositoryReconciler) ImageRepositoryForSameUrlExists(ctx context.Context, imageRepository *imagerepositoryv1alpha1.ImageRepository) (bool, error) {
+	log := ctrllog.FromContext(ctx)
+	imageRepositoriesList := &imagerepositoryv1alpha1.ImageRepositoryList{}
+	if err := r.Client.List(ctx, imageRepositoriesList, &client.ListOptions{Namespace: imageRepository.Namespace}); err != nil {
+		log.Error(err, "failed to list image repositories")
+		return false, err
+	}
+
+	imageRepositoryUrl := imageRepository.Status.Image.URL
+	imageRepositoryName := imageRepository.ObjectMeta.Name
+	imageRepositoryFound := false
+	for _, imageRepo := range imageRepositoriesList.Items {
+		if imageRepositoryUrl == imageRepo.Status.Image.URL {
+			if imageRepositoryName == imageRepo.ObjectMeta.Name {
+				continue
+			}
+			imageRepositoryFound = true
```

Comment:
We don't need `imageRepositoryFound` variable at all, just return here

Reply:
done